src/Nethermind/Nethermind.JsonRpc/Modules/Trace/TraceRpcModule.cs

```diff
@@ -59,20 +56,20 @@ public static ParityTraceTypes GetParityTypes(string[] types) =>
         /// <summary>
         /// Traces one transaction. Doesn't charge fees.
         /// </summary>
-        public ResultWrapper<ParityTxTraceFromReplay> trace_call(TransactionForRpc call, string[] traceTypes, BlockParameter? blockParameter = null)
+        public ResultWrapper<ParityTxTraceFromReplay> trace_call(TransactionForRpc call, string[] traceTypes, BlockParameter? blockParameter = null, Dictionary<Address, AccountOverride>? stateOverride = null)
```

Comment:
`eth_call`, and `eth_estimategas` are actually the only methods where state override is supported in [Geth](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-eth#eth-call), and [some popular RPC providers](https://www.quicknode.com/docs/ethereum/eth_estimateGas).

While this PR extends it to other methods, there is a risk of a future signature mismatch, in case other clients update them differently. For example, add `options` as the next parameter, similar to `debug_traceCall`.

For these cases, it may be better to introduce new methods instead (like `nethermind_traceCall` or `trace_callv2`) with the same logic, but a new parameter added. and leave old ones intact.

Reply:
Can you check what Reth does? If it doesn't add it than maybe we shouldn't either.
Other than that I wouldn't sweat about it, those methods are stable for years now

Reply:
Yeah, that did add it to some other methods apart from 2 Geth does:
- [eth](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-eth-api/src/core.rs): `eth_call`, `eth_callMany`, `eth_estimateGas` (but not `eth_createAccessList` as requested in one of the linked issues)
- [trace](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-api/src/trace.rs): just for `trace_call` (not for `trace_call_many` for some reason)
- [debug](https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-api/src/debug.rs): `debug_trace_call`, `debug_trace_call_many` (in all cases nested in 3rd `options` parameter)

Reply:
let's mimic Reth api then